r/R/query-engine.R

```diff
@@ -165,6 +172,12 @@ ExecPlan <- R6Class("ExecPlan",
             left_suffix = .data$join$suffix[[1]],
             right_suffix = .data$join$suffix[[2]]
           )
+          # If we are doing a full outer join and not keeping the join keys of
```

Comment:
Would it make sense to move this projection step into `full_join.arrow_dplyr_query()`? I could see arguments either way, but it might be cleaner to move the logic there so that there is less special handling in `ExecPlan$build()`, so that it can more simply just render the query plan it is given.

Reply:
We could do that, but I would have to yank `left_output` and `right_output` resolution up to `do_join`, since those are coupled with `keep` and the full outer coalesce.

Reply:
I just pushed a refactor commit that does that. Let me know what you think.